File: C++/Organizm.cpp
#include "Organizm.h"

Organizm::Organizm(int x, int y, int sila, int inicjatywa, int id, Swiat* swiat)
{
	this->id = id;
	this->swiat = swiat;
	this->x = x;
	this->y = y;
	this->sila = sila;
	this->inicjatywa = inicjatywa;
	this->dopiero_zrodzony = false;
	this->rundy = 0;
	this->swiat->SetOrganizmy(this->GetX(), this->GetY(), this);
	swiat->addToList(this);
}

Organizm::Organizm(int x, int y, int sila, int inicjatywa, int id, bool dz, Swiat* swiat)
{
	this->id = id;
	this->swiat = swiat;
	this->x = x;
	this->y = y;
	this->sila = sila;
	this->inicjatywa = inicjatywa;
	this->dopiero_zrodzony = dz;
	this->rundy = 0;
	this->swiat->SetOrganizmy(this->GetX(), this->GetY(), this);
	swiat->addToList(this);
}

int Organizm::CzyOdbijaAtak(Organizm* atakujacy)
{
	return 0;
}

bool Organizm::CzyWech(Organizm* organizm)
{
	return true;
}

bool Organizm::CzyDodajeSily()
{
	return false;
}

bool Organizm::CzyUcieka()
{
	return false;
}

bool Organizm::CzyNieZabija()
{
	return true;
}

void Organizm::idzXY(int x, int y)
{
	this->swiat->SetOrganizmy(this->GetX(), this->GetY(), nullptr);
	this->SetX(x);
	this->SetY(y);
	this->swiat->SetOrganizmy(this->GetX(), this->GetY(), this);
}

int Organizm::GetY() const
{
	return y;
}
int Organizm::GetX() const
{
	return x;
}
int Organizm::GetSila() const
{
	return sila;
}
int Organizm::GetInicjatywa() const
{
	return inicjatywa;
}
int Organizm::GetId() const
{
	return id;
}
bool Organizm::GetDZ() const
{
	return dopiero_zrodzony;
}
void Organizm::SetX(int x)
{
	this->x = x;
}
void Organizm::SetY(int y)
{
	this->y = y;
}
void Organizm::SetSila(int sila)
{
	this->sila = sila;
}
void Organizm::SetInicjatywa(int inicjatywa)
{
	this->inicjatywa = inicjatywa;
}
void Organizm::SetDZ(bool zrodzony)
{
	this->dopiero_zrodzony = zrodzony;
}


void Organizm::WyzerujRundy()
{
	this->rundy = 0;
}
void Organizm::KolejnaRunda()
{
	this->rundy++;
}
int Organizm::GetRundy()
{
	return this->rundy;
}

void Organizm::SetRundy(int r)
{
	this->rundy = r;
}

std::string Organizm::UczestnikWydarzenia()
{
	std::string result = "";

	result = this->nazwa();

	result += "(";
	result += std::to_string(this->GetX());
	result += ",";
	result += std::to_string(this->GetY());
	result += ")";

	return result;
}

Organizm::~Organizm()
{
}

File: Python/Organizm.py
from abc import ABC, abstractmethod


class Organizm(ABC):
    def __init__(self, x, y, sila, inicjatywa, idi, dz, swiat):
        self.id = idi
        self.swiat = swiat
        self.x = x
        self.y = y
        self.sila = sila
        self.inicjatywa = inicjatywa
        self.dopiero_zrodzony = dz
        self.rundy = 0
        self.swiat.SetOrganizmy(self.GetX(), self.GetY(), self)
        swiat.addToList(self)

    @abstractmethod
    def akcja(self):
        pass

    @abstractmethod
    def kolizja(self, organizm, kierunek):
        pass

    @abstractmethod
    def rysowanie(self):
        pass

    @abstractmethod
    def nazwa(self):
        pass

    @abstractmethod
    def UczestnikWydarzenia(self):
        pass

    def UczestnikWydarzenia(self):
        result = ""

        result = self.nazwa()

        result += "("
        result += str(self.GetX())
        result += ","
        result += str(self.GetY())
        result += ")"

        return result

    def CzyOdbijaAtak(self, atakujacy):
        return 0

    def CzyWech(self, organizm):
        return True

    def CzyDodajeSily(self):
        return False

    def CzyUcieka(self):
        return False

    def CzyNieZabija(self):
        return True

    def idzXY(self, x, y):
        self.swiat.SetOrganizmy(self.GetX(), self.GetY(), None)
        self.SetX(x)
        self.SetY(y)
        self.swiat.SetOrganizmy(self.GetX(), self.GetY(), self)

    def GetX(self):
        return self.x

    def GetY(self):
        return self.y

    def GetSila(self):
        return self.sila

    def GetInicjatywa(self):
        return self.inicjatywa

    def GetDZ(self):
        return self.dopiero_zrodzony

    def GetId(self):
        return self.id

    def SetX(self, x):
        self.x = x

    def SetY(self, y):
        self.y = y

    def SetSila(self, sila):
        self.sila = sila

    def SetInicjatywa(self, inicjatywa):
        self.inicjatywa = inicjatywa

    def SetDZ(self, zrodzony):
        self.dopiero_zrodzony = zrodzony

    def WyzerujRundy(self):
        self.rundy = 0

    def SetRundy(self, r):
        self.rundy = r

    def KolejnaRunda(self):
        self.rundy += 1

    def GetRundy(self):
        return self.rundy



